app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -4,8 +4,36 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
   require_authentication :all, scopes: [:project]
   schema_type :json_schema
 
+  def update_project_settings
+    binding.pry
+    project = Project.find params[:project_id]
+    unless required_params
+      render json: { error: 'missing or invalid params' }, status: :unprocessable_entity
+    end
+    if project.owner?(api_user.user)
```

Comment:
By adding a new "settings" attribute that's intended to be fully editable by project owners, we can leave "preferences" locked to the user, along with sensitive stuff like mail prefs.